cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesAdder.java

```diff
@@ -127,22 +326,115 @@ private static PropertyBag createPowerFlowProperties(CgmesModel cgmes, Terminal
+                topologicalIsland.put(CgmesNames.ANGLEREF_TOPOLOGICALNODE,
+                    value.get(0).getId(CgmesNames.ANGLEREF_TOPOLOGICALNODE));
+                topologicalIsland.put(CgmesNames.TOPOLOGICAL_NODES,
+                    getMultivaluedProperty(value, CgmesNames.TOPOLOGICAL_NODES));
+                topologicalIslands.add(topologicalIsland);
+            });
+            cgmes.add(originalSVcontext, CgmesNames.TOPOLOGICAL_ISLAND, topologicalIslands);
+        }
+    }
+
+    // Added full model data with proper profile (StateVariables)
+    // FullModel is defined in ModelDescription:
+    // http://iec.ch/TC57/61970-552/ModelDescription/1#
+    private void addModelDescriptionToCgmes() {
+        PropertyBags fullModelSV = new PropertyBags();
+        // for properties such as "md:Model.DependentOn" which might have arbitrary
+        // number of values, SPARQL will return multiple result sets.
+        // All are equal, except the multiValued property value.
+        if (!originalFullModel.isEmpty()) {
+            PropertyBag newModelDescription = new PropertyBag(SV_FULLMODEL_PROPERTIES);
+            newModelDescription
+                .setClassPropertyNames(
+                    Arrays.asList(CgmesNames.SCENARIO_TIME, CgmesNames.CREATED, CgmesNames.DESCRIPTION,
+                        CgmesNames.VERSION, CgmesNames.DEPENDENT_ON, CgmesNames.PROFILE,
+                        CgmesNames.MODELING_AUTHORITY_SET));
+            newModelDescription.setMultivaluedProperty(Arrays.asList(CgmesNames.DEPENDENT_ON));
+            newModelDescription.put(CgmesNames.SCENARIO_TIME, originalFullModel.get(0).getId("scenarioTime"));
+            newModelDescription.put(CgmesNames.CREATED, originalFullModel.get(0).getId("created"));
+            newModelDescription.put(CgmesNames.DESCRIPTION, originalFullModel.get(0).getId("description"));
+            newModelDescription.put(CgmesNames.VERSION, originalFullModel.get(0).getId("version"));
+            newModelDescription.put(CgmesNames.DEPENDENT_ON,
+                getMultivaluedProperty(originalFullModel, "DependentOn"));
+            newModelDescription.put(CgmesNames.PROFILE, originalFullModel.get(0).getId("profile"));
+            newModelDescription.put(CgmesNames.MODELING_AUTHORITY_SET,
+                originalFullModel.get(0).getId("modelingAuthoritySet"));
+            fullModelSV.add(newModelDescription);
+            cgmes.add(originalSVcontext, CgmesNames.FULL_MODEL, fullModelSV);
+        }
+    }
+
+    private String getMultivaluedProperty(PropertyBags pb, String multivaluedPropertyName) {
+        // for properties such as "md:Model.DependentOn" which might have arbitrary
+        // number of values, SPARQL will return multiple result sets. We will loop
+        // through all and collect all values for multiValued property.
+        List<String> list = new ArrayList<>();
+        pb.forEach(m -> list.add(m.getId(multivaluedPropertyName)));
```

Comment:
Use the stream API to initialize `list`. If there is no stream() method in PropertyBags, you should consider to add one.

Reply:
Done. PropertyBags.pluckLocals(String property) does what we need, I missed it initially.